tests/src/Api/FileIngestionApiTest.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenEuropa\Tests\EuropaSearchClient\Api;
+
+use GuzzleHttp\Psr7\Response;
+use OpenEuropa\EuropaSearchClient\Model\Ingestion;
+use OpenEuropa\Tests\EuropaSearchClient\Traits\ClientTestTrait;
+use PHPUnit\Framework\TestCase;
+
+/**
+ * @coversDefaultClass \OpenEuropa\EuropaSearchClient\Api\FileIngestionApi
+ */
+class FileIngestionApiTest extends TestCase
+{
+    use ClientTestTrait;
+
+    /**
+     * @covers ::ingest
+     * @dataProvider providerTestFileIngestion
+     *
+     * @param array $clientConfig
+     * @param array $responses
+     * @param mixed $expectedResult
+     */
+    public function testFileIngestion(array $clientConfig, array $responses, $expectedResult): void
+    {
+        $actualResult = $this->getTestingClient($clientConfig, $responses)
+            ->ingestFile('http://example.com', chr(0) . 'saDD*&6778)(&88');
```

Comment:
Would it be worth here passing the binary of a real file (adding one in a fixtures folders and then file_get_contents..) ?